router_api.go

```diff
@@ -15,8 +31,17 @@ func newAPIHandler(rout *Router) http.Handler {
 			w.WriteHeader(http.StatusMethodNotAllowed)
 			return
 		}
-
-		rout.ReloadRoutes()
+		// Send a message to the reload goroutine, which will start a new timeout
+		// before reloading, or do nothing if one is already in progress.
+		select {
+		case reloadChan <- true:
+			logInfo("router: reload triggered")
+			w.WriteHeader(http.StatusAccepted)
+			w.Write([]byte("Reload triggered"))
+		default:
+			logInfo("router: reload already in progress")
+			w.Write([]byte("Reload already in progress"))
+		}
```

Comment:
The problem here is that it's using an unbuffered channel, so a send on the channel will only succeed when there's something waiting to receive - https://golang.org/ref/spec#Send_statements.